Stratis.Bitcoin/Features/MemoryPool/MemPoolCoinView.cs

```diff
@@ -9,35 +9,103 @@
 
 namespace Stratis.Bitcoin.Features.MemoryPool
 {
+    /// <summary>
+    /// Memory pool coin view.
+    /// Provides coin view representation of memory pool transactions via a backed coin view.
+    /// </summary>
```

Comment:
When I first learned about this code base I did not know what coin view was. In Bitcoin, I think people understand what UTXOs are, but I never heard about "coin view", so it might be a good idea to use that Bitcoin terminology when commenting. However, I admit that I am not familiar with Bitcoin Core code base, so maybe "coin view" is term from there and I am just ignorant of there, in which case feel free to ignore me :)